rust/src/io/commit.rs

```diff
@@ -68,10 +68,17 @@ pub trait CommitHandler: Debug + Send + Sync {
         path: &Path,
         object_store: &ObjectStore,
     ) -> std::result::Result<u64, crate::Error> {
+        // check existence first
```

Comment:
Does `read_manifest` not return a NotFound error if not found? Just making sure we aren't adding a new IO call for no reason.

Reply:
it returns `crate::Error::IO` sadly